packages/jest-config/src/getMaxWorkers.ts

```diff
@@ -18,18 +18,18 @@ export default function getMaxWorkers(
     const maxWorkers = (argv.maxWorkers as unknown) as number | string;
     const parsed = parseInt(maxWorkers as string, 10);
 
+    const cpus = os.cpus().length;
     if (
       typeof maxWorkers === 'string' &&
       maxWorkers.trim().endsWith('%') &&
       parsed > 0 &&
       parsed <= 100
     ) {
-      const cpus = os.cpus().length;
       const workers = Math.floor((parsed / 100) * cpus);
       return workers >= 1 ? workers : 1;
     }
 
-    return parsed > 0 ? parsed : 1;
+    return Math.max(Math.min(cpus, parsed), 1);
```

Comment:
I think we should use cpus - 1, which is usually the default. If you really want to use even the last core for a worker and not the main process / other processes, you can still do `'100%'`